src/com/griddynamics/main/Triangle.java

```
@@ -0,0 +1,69 @@
+        final double EPS = 0.00001;
+        return Math.abs(a - b) < EPS;
+    }
+
+    public static boolean checkIfRectangular(double x, double y, double z) {
+        return doubleEquals(x * x + y * y, z * z) || doubleEquals(x * x + z * z, y * y) || doubleEquals(z * z + y * y, x * x);
+    }
+
+    public static boolean checkIfIsosceles(double x, double y, double z) {
+        return doubleEquals(x, y) || doubleEquals(x, z) || doubleEquals(y, z);
+    }
+
+    public static TriangleType triangleAnalysisSystem(double x, double y, double z) {
+        if (!(doubleEquals(x, 0) || doubleEquals(y, 0) || doubleEquals(z, 0)) && x + y > z && x + z > y && y + z > x) {
+            if (doubleEquals(x, y) && doubleEquals(x, z)) {
+                System.out.println("You have equilateral triangle");
+                return TriangleType.EQUILATERAL_TRIANGLE;
+            } else {
+                if (checkIfRectangular(x, y, z)) {
+                    if (checkIfIsosceles(x, y, z)) {
+                        System.out.println("You have rectangular isosceles triangle");
+                        return TriangleType.RECTANGULAR_ISOSCELES_TRIANGLE;
+                    } else {
+                        System.out.println("You have rectangular triangle");
+                        return TriangleType.RECTANGULAR_TRIANGLE;
+                    }
+                } else {
+                    if (checkIfIsosceles(x, y, z)) {
+                        System.out.println("You have isosceles triangle");
+                        return TriangleType.ISOSCELES_TRIANGLE;
+                    } else {
+                        System.out.println("You have non-sided triangle");
+                        return TriangleType.DEFAULT_TRIANGLE;
+                    }
+                }
+            }
+        } else {
+            System.out.println("Sorry, it is impossible to make a triangle from your data");
+            return TriangleType.NOT_TRIANGLE;
```

@kolegova-mary the same as for lucky ticket, let's throw an exception instead of returning enum